[BLU AF3] Implement Nepionic Soulflayer NM

hooksta4 opened the pull request:
Would like some assistance on the mobskills (mind and shield) as well as verifying the spell list. I used the spell list that was on bg/cylopedia, watched a few videos and gave it some more HP as it was just too low.

Thanks!

<!-- remove space and place 'x' mark between square [] brackets or click the checkbox after saving to affirm: -->
**_I affirm:_**
- [x] I have paid attention to this example and will edit again if need be to not break the formatting, or I will be ignored
- [x] that I agree to LandSandBoat's [Limited Contributor License Agreement](https://github.com/LandSandBoat/server/blob/base/.github/CONTRIBUTOR_AGREEMENT.md), as written on this date
- [x] that I have **read the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md)**
- [x] that I've _**tested my code and things my code changed**_ since the last commit in the PR, and will test after any later commits

hooksta4 committed:
[WIP] BLU AF Quest "Transformations" #1061

Would like some assistance on the mobskills (mind and shield) as well as verifying the spell list. I used the spell list that was on bg/cylopedia, watched a few videos and gave it some more HP as it was just too low.

Thanks!

Xaver-DaRed commented on scripts/globals/mobskills/Immortal_mind.lua:
```diff
@@ -0,0 +1,43 @@
+        mabTotal = mabTotal:getPower() + 10
+    else
+        mabTotal = 10
+    end
```
Why not...
```
local mabTotal = 10
if mob:getStatusEffect(xi.effect.MAGIC_ATK_BOOST) ~= nil then
    mabTotal = mabTotal:getPower() + 10
end
```
Or something of the short...
Also, uneeded parenthesis

TeoTwawki replied:
that get status already in a variable above. use that instead of executing another get

TeoTwawki replied:
mabTotal was defined as the effect itself, not its power. you're are effectively saying effect+10 instead of effect power +10

TeoTwawki replied:
`getStatusEffect()` gets the effect itself.. you want to get the power from that effect using `effect:getPower()` where effect is the effect entity, not its ID (so chain get status into get power)

TeoTwawki replied:
pseudo code, do not copypaste:
```lua
local effectID = xi.effect.dunnoLol
local effect = GetStatusEffect(effectID)
local effectPower = effect:getPower()
local powerNow - 0
if effect then
    powernow = effectpower + 10
end
```

TeoTwawki replied:
pseudo code, do not copypaste:
```lua
local effectID = xi.effect.dunnoLol
local effect = mob:getStatusEffect(effectID)
local effectPower = effect:getPower()
local powerNow - 0
if effect then
    powernow = effectpower + 10
end
```

claywar commented on scripts/globals/mobskills/Immortal_mind.lua:
```diff
@@ -0,0 +1,43 @@
+    else
+        mdbTotal = 10
+    end
+    -- print(mabTotal)
```
Could you please remove these commented debug lines?

claywar commented on scripts/globals/mobskills/Immortal_shield.lua:
```diff
@@ -0,0 +1,28 @@
+local mobskill_object = {}
+
+mobskill_object.onMobSkillCheck = function(target, mob, skill)
+
```
Unnecessary newline

TeoTwawki replied:
actually he's missing the return that should be here

TeoTwawki replied:
You planning to come back to this PR @hooksta4 ? without the return line this script will error out

hooksta4 replied:
Yes, started a new job and been doing shift work. Will get to this as soon as I can.

claywar commented on scripts/zones/Alzadaal_Undersea_Ruins/IDs.lua:
```diff
@@ -54,7 +54,7 @@ zones[xi.zone.ALZADAAL_UNDERSEA_RUINS] =
         RUNIC_PORTAL_NORTH      = 17072236,
         RUNIC_PORTAL_SOUTH      = 17072237,
-        NEPIONIC_QM             = 17072271,
+        NEPIONIC_QM             = 17072378,
```
I'm fairly certain I just fixed this in latest NPC shifts

hooksta4 replied:
Yea I didn't realize there was a new push after I made this branch :(

TeoTwawki replied:
_all your rebase are belong to us_

hooksta4 committed:
Updated spell_list ID and Alzadaal Ruins IDs.lua

hooksta4 committed:
Merge branch 'base' into transformations

hooksta4 commented:
very weird, I made some changes and loaded up the test server to try it all out and i keep getting black screened on CS ID 4... gonna take me a bit to figure this one out =/

hooksta4 committed:
Merge branch 'base' into transformations

hooksta4 committed:
Merge branch 'LandSandBoat:base' into transformations

hooksta4 committed:
Merge branch 'transformations' of https://github.com/hooksta4/LSB_Hook_fork into transformations

hooksta4 committed:
Merge branch 'LandSandBoat:base' into transformations

hooksta4 committed:
Merge branch 'transformations' of https://github.com/hooksta4/LSB_Hook_fork into transformations

hooksta4 committed:
Updated Immortal_mind.lua and Immortal_shield.lua

Update to mob_skills.sql, and Immortal_mind/shield luas to reflect comments from above. Pulled up on test server, confirmed that all skills and spells were used and didnt break the server. Specifically to, the shield reduced gust slash damage when it was active, logs also showed removed and went back to normal gust slash damage. All other CS's work to include obtaining AF hat.

Xaver-DaRed commented on sql/mob_groups.sql:
```diff
@@ -3355,7 +3355,7 @@ INSERT INTO `mob_groups` VALUES (1,3145,71,'Pit_Monster',0,128,0,0,0,0,0,0);
 -- ------------------------------------------------------------
 
-INSERT INTO `mob_groups` VALUES (1,2834,72,'Nepionic_Soulflayer',0,128,0,0,0,60,60,0);
+INSERT INTO `mob_groups` VALUES (1,2834,72,'Nepionic_Soulflayer',0,128,0,6800,0,60,60,0);
```
Wouldnt this be 2834?

hooksta4 replied:
I thought that was the mobs HP?

Xaver-DaRed replied:
very ture, my bad

Skerxan commented:
What's the status of this PR? Can someone review please?
